haystack/components/preprocessors/recursive_splitter.py

```diff
@@ -343,11 +389,15 @@ def _fall_back_to_fixed_chunking(self, text: str, split_units: Literal["word", "
 
             if current_chunk:
                 chunks.append("".join(current_chunk))
-
-        else:
-            for i in range(0, self._chunk_length(text), step):
+        elif split_units == "char":
+            for i in range(0, self._chunk_length(text), self.split_length):
                 chunks.append(text[i : i + self.split_length])
-
+        else:  # token
+            assert self.tiktoken_tokenizer is not None  # mypy doesn't know that warm_up() is called
```

Comment:
I would remove his assert, it's only here due to the limitations of the type checker and therefore I would just add a comment  and then below a # type: ignore

```suggestion
            # at this point we know that the tokenizer is already initialized
```